libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRoutePreviewScreen.kt

```diff
@@ -66,6 +67,7 @@ class CarRoutePreviewScreen @UiThread constructor(
         override fun handleOnBackPressed() {
             logAndroidAuto("CarRoutePreviewScreen OnBackPressedCallback")
             MapboxNavigationApp.current()!!.setNavigationRoutes(emptyList())
+            MapboxScreenManager.clear()
```

Comment:
After you press back in `CarRoutePreviewScreen `, `MapboxScreenManager.screenEvent.value` becomes null, so it doesn't represent the screen that is currently opened, should we fix this? Even worse, after you press back on any other screen, `MapboxScreenManager.screenEvent.value` still references the screen that you closed.

Reply:
Yeah I didn't like this either. But I also found issues with adding a backstack to `MapboxScreenManager`, because that will easily become inconsistent with the back stack in `ScreenManager`. I was looking for a way to observe the backstack but it is not available. (it is worth looking into a MapboxScreenManager factory backstack again to handle this).

Adding the clear seems reasonable because an assumption I'm making is that developers will use the jetpack `ScreenManager` for some cases, and I'm making it possible for them to `clear` the  `MapboxScreenManager` in those cases.

There are alternatives to how this specific case is handled. For example the onBack could replaceTop to FreeDrive, which is what I tried at first. The issue is that RoutePreview can start from PlaceListOnMap or from PlaceSearch.

Reply:
I'll add a `MapboxScreenOperation.GO_BACK` and see what it ends up looking like

Reply:
This event is tricky

Reply:
OK it works! It's not the most elegant solution but it has a lot of tests and it seems like it will be useful.

Now you can call `mapboxScreenManager.goBack()`, and observe the screen change through `MapboxScreenManager.screenEvent`

Reply:
Resolving this thread, and can make new discussions on it